Scenes/Outline.shader

```diff
@@ -0,0 +1,56 @@
+shader_type spatial;
+render_mode unshaded;
+
+uniform float scale = 2.0;
+uniform float threshold = 2.0;
+uniform vec3 color = vec3(0, 0, 0);
+uniform bool enabled = true;
+
+varying mat4 CAMERA;
+
+void vertex() 
+{
+	POSITION = vec4(VERTEX, 1.0);
+	CAMERA = CAMERA_MATRIX;
+}
+
+float depth_calc(vec2 uv, sampler2D depth_tex, mat4 inv_projection_mat) 
+{
+	// based on: https://docs.godotengine.org/en/stable/tutorials/shading/advanced_postprocessing.html
+	float d = texture(depth_tex, uv).r;
+	vec3 ndc = vec3(uv, d) * 2.0 - 1.0;
+	vec4 view = inv_projection_mat * vec4(ndc, 1.0);
+  	view.xyz /= view.w;
+	// For future use: computing world coordinates
+	// vec4 world = CAMERA * inv_projection_mat * vec4(ndc, 1.0);
+  	// vec3 world_position = world.xyz / world.w;
+  	float linear_depth = -view.z;
+	return linear_depth;
+}
+
+float diff_sq(vec2 uv, vec2 dir, sampler2D depth_tex, mat4 inv_projection_mat) {
```

Comment:
if use `DEPTH_TEXTURE` directly then no need to have the arguments here too

Reply:
For some reason cannot access from the function. Quite strange.